eng/install-native-dependencies.sh

```diff
@@ -13,37 +15,35 @@
 #
 # ./install-native-dependencies.sh <OS>
 
-if [ "$1" = "Linux" ]; then
-    sudo apt update
-    if [ "$?" != "0" ]; then
-       exit 1;
-    fi
-    sudo apt install cmake llvm-3.9 clang-3.9 lldb-3.9 liblldb-3.9-dev libunwind8 libunwind8-dev gettext libicu-dev liblttng-ust-dev libcurl4-openssl-dev libssl-dev libkrb5-dev libnuma-dev build-essential
-    if [ "$?" != "0" ]; then
-        exit 1;
-    fi
-elif [[ "$1" == "MacCatalyst" || "$1" == "OSX" || "$1" == "tvOS" || "$1" == "iOS" ]]; then
-    engdir=$(dirname "${BASH_SOURCE[0]}")
+os="$(echo "$1" | tr "[:upper:]" "[:lower:]")"
+
+if [ -e /etc/os-release ]; then
+    . /etc/os-release
+fi
 
-    echo "Installed xcode version: `xcode-select -p`"
+if [ "$os" = "linux" ] && { [ "$ID" = "debian" ] || [ "$ID_LIKE" = "debian" ]; }; then
+    apt update
+
+    apt install -y build-essential gettext locales cmake llvm clang lldb liblldb-dev libunwind8-dev libicu-dev liblttng-ust-dev \
+        libssl-dev libkrb5-dev libnuma-dev
+
+    localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
+elif [ "$os" = "maccatalyst" ] || [ "$os" = "osx" ] || [ "$os" = "macos" ] || [ "$os" = "tvos" ] || [ "$os" = "ios" ]; then
+    echo "Installed xcode version: $(xcode-select -p)"
 
     if [ "$3" = "azDO" ]; then
         # workaround for old osx images on hosted agents
         # piped in case we get an agent without these values installed
-        if ! brew_output="$(brew uninstall openssl@1.0.2t 2>&1 >/dev/null)"; then
+        if ! brew uninstall openssl@1.0.2t >/dev/null 2>&1; then
```

Comment:
Do you know if this issue still applies to the current osx images on our hosted agents?

Reply:
Not sure if the workaround is really needed, but `azDO` argument is in use:

```sh
$ git grep install-native-dependencies eng/pipelines 

eng/pipelines/common/global-build-job.yml:      - script: $(Build.SourcesDirectory)/eng/install-native-dependencies.sh ${{ parameters.osGroup }} ${{ parameters.archType }} azDO
eng/pipelines/common/templates/runtimes/build-test-job.yml:      - script: $(Build.SourcesDirectory)/eng/install-native-dependencies.sh $(osGroup) ${{ parameters.archType }} azDO
eng/pipelines/coreclr/templates/build-jit-job.yml:      - script: $(Build.SourcesDirectory)/eng/install-native-dependencies.sh $(osGroup) ${{ parameters.archType }} azDO
eng/pipelines/coreclr/templates/build-job.yml:      - script: $(Build.SourcesDirectory)/eng/install-native-dependencies.sh $(osGroup) ${{ parameters.archType }} azDO
eng/pipelines/installer/jobs/build-job.yml:        - script: $(Build.SourcesDirectory)/eng/install-native-dependencies.sh ${{ parameters.osGroup }} ${{ parameters.archType }} azDO
eng/pipelines/libraries/build-job.yml:          - script: $(Build.SourcesDirectory)/eng/install-native-dependencies.sh ${{ parameters.osGroup }} ${{ parameters.archType }} azDO
eng/pipelines/mono/templates/build-job.yml:      - script: $(Build.SourcesDirectory)/eng/install-native-dependencies.sh $(osGroup) ${{ parameters.archType }} azDO
eng/pipelines/mono/templates/generate-offsets.yml:      - script: $(Build.SourcesDirectory)/eng/install-native-dependencies.sh $(osGroup) ${{ parameters.archType }} azDO
```

Reply:
As far as I know this issue no longer happens on the macOS hosted agents so we can try removing it. I don't have a preference for whether you want to do it as part of this PR or separately